### PR TITLE
Update examples to use new Swift translation target

### LINKED ISSUE #173: Use modern archiving and unarchiving methods in GTMAppAuthFetcherAuthorization.swift
Due to some archiving bugs, we had to revert back to the older methods for archiving and unarchiving `GTMAppAuthFetcherAuthorization` in Swift. This reversion should only be temporary as there is a fix underway. When ready, update the following:

* `GTMAppAuthFetcherAuthorization.save(authorization:itemName:)` to use `NSKeyedArchiver.archivedData(withRootObject:requiringSecureCoding:)`
* Uncomment the code checking for version availability (to call the modern unarchiving method) in `GTMAppAuthFetcherAuthorizaiton.authorization(itemName:)`

Both of these have commented out code and a link to this issue.

### PR DESCRIPTION
**Pull request by mdmathias**
We should close #173 before merging this branch to main.

### CODE REVIEW

File: Examples/Example-iOS/Source/GTMAppAuthExampleViewController.m [hunk 1]
```diff
@@ -39,7 +39,7 @@
         https://console.developers.google.com/apis/credentials?project=_
         The client should be registered with the "iOS" type.
  */
-static NSString *const kClientID = @"YOUR_CLIENT.apps.googleusercontent.com";
+static NSString *const kClientID = @"687389107077-8qr6dh8fr4uaja89sdr5ieqb7mep04qv.apps.googleusercontent.com";
```
Comment (petea) [inline Examples/Example-iOS/Source/GTMAppAuthExampleViewController.m:42]: Is this a test project of yours?

File: Examples/Example-iOS/Source/GTMAppAuthExampleViewController.m [hunk 2]
```diff
@@ -107,19 +107,29 @@ - (void)viewDidLoad {
 /*! @brief Saves the @c GTMAppAuthFetcherAuthorization to @c NSUSerDefaults.
  */
 - (void)saveState {
+  NSError *error;
   if (_authorization.canAuthorize) {
     [GTMAppAuthFetcherAuthorization saveAuthorization:_authorization
-                                    toKeychainForName:kExampleAuthorizerKey];
+                                         withItemName:kExampleAuthorizerKey
+                                                error:&error];
```
Comment (petea) [inline Examples/Example-iOS/Source/GTMAppAuthExampleViewController.m:114]: Didn't we decide _not_ to expose the error handling improvements in the public API at this point?

**Review by petea — COMMENTED** [review]
How about the pod projects?

File: Examples/Example-iOS/Source/GTMAppAuthExampleViewController.m (hunk 1 above)
Comment (mdmathias) [reply Examples/Example-iOS/Source/GTMAppAuthExampleViewController.m:42]: It is. I will remove before review (converted this PR back to draft).

File: Examples/Example-iOS/Source/GTMAppAuthExampleViewController.m (hunk 2 above)
Comment (mdmathias) [reply Examples/Example-iOS/Source/GTMAppAuthExampleViewController.m:114]: Let's talk more about this! I think I'd prefer to leave the error handling in place to help this PR move forward, but I'd be happy to make changes to this API before we land the `swift-translation` branch.

File: Examples/Example-iOS/Podfile [hunk 3]
```diff
@@ -2,14 +2,14 @@ platform :ios, '9.0'
...
   # Pods for GTMAppAuth development
-  pod 'GTMAppAuth', :path => '../../'
+  pod 'GTMAppAuthSwift', :path => '../../'
 
   # In production, you would use:
   # pod 'GTMAppAuth'
 
-  pod 'AppAuth', '~> 1.0'
-  pod 'GTMSessionFetcher/Core', '~> 1.0'
+  pod 'AppAuth', '~> 1.4'
```
Comment (petea) [inline Examples/Example-iOS/Podfile:13]: Do we need to bump these two deps?  If not, I'd leave them unchanged.

File: GTMAppAuthSwift/Sources/GTMAppAuthFetcherAuthorization.swift [hunk 4]
```diff
@@ -15,8 +15,13 @@
  */
 
 import Foundation
+#if SWIFT_PACKAGE
```
Comment (petea) [inline GTMAppAuthSwift/Sources/GTMAppAuthFetcherAuthorization.swift:18]: Should we add a comment explaining why we have to do this?

File: GTMAppAuthSwift/Sources/GTMOAuth2KeychainCompatibility.swift [hunk 5]
```diff
@@ -15,8 +15,13 @@
  */
 
 import Foundation
+#if SWIFT_PACKAGE
```
Comment (petea) [inline GTMAppAuthSwift/Sources/GTMOAuth2KeychainCompatibility.swift:18]: Same as above.

File: GTMAppAuthSwift/Sources/GTMAppAuthFetcherAuthorization.swift [hunk 6]
```diff
@@ -15,8 +15,14 @@
  */
 
 import Foundation
+// This check is needed to ensure that we import the correct dependency for both SPM and CocoaPods
```
Comment (petea) [inline GTMAppAuthSwift/Sources/GTMAppAuthFetcherAuthorization.swift:18]: Might add something along the lines of "CocoaPods doesn't define separate Clang modules for subspecs."

**Review by petea — APPROVED** [review]

**Pull request merged**